src/Microsoft.AspNetCore.Razor.Language/RazorLanguageVersion.cs

```diff
@@ -1,33 +1,33 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
+
 namespace Microsoft.AspNetCore.Razor.Language
 {
-    public enum RazorLanguageVersion
+    public sealed class RazorLanguageVersion
     {
-        Version1_0 = 1,
+        public static readonly RazorLanguageVersion Version1_0 = new RazorLanguageVersion(1, 0);
 
-        Version1_1 = 2,
+        public static readonly RazorLanguageVersion Version1_1 = new RazorLanguageVersion(1, 1);
 
-        Version2_0 = 3,
+        public static readonly RazorLanguageVersion Version2_0 = new RazorLanguageVersion(2, 0);
 
-        Version2_1 = 4,
-    }
+        public static readonly RazorLanguageVersion Version2_1 = new RazorLanguageVersion(2, 1);
 
-    internal static class RazorLanguageVersionExtensions
-    {
-        internal static bool IsValid(this RazorLanguageVersion version)
+        public static readonly RazorLanguageVersion Latest = Version2_1;
+
+        // Don't want anyone else constructing language versions.
+        private RazorLanguageVersion(int major, int minor)
         {
-            switch (version)
-            {
-                case RazorLanguageVersion.Version1_0:
-                case RazorLanguageVersion.Version1_1:
-                case RazorLanguageVersion.Version2_0:
-                case RazorLanguageVersion.Version2_1:
-                    return true;
-            }
-
-            return false;
+            Major = major;
+            Minor = minor;
         }
+
+        public int Major { get; }
+
+        public int Minor { get; }
+
+        public override string ToString() => $"Razor Language '{Major}.{Minor}'";
     }
```

Comment:
My vision for this is that when we eventually flow the MSBuild Razor Language version we'll add a method on here to determine or create a proper `RazorLanguageVersion`. Either way I wanted to avoid us having a potentially invalid `RazorLanguageVersion` in our system and therefore having to check `IsValid` in several locations (that's why this is the only class that can construct this class).

Reply:
Yeah, this will eventually have a `TryParse` method.